runtime/compiler/codegen/J9TreeEvaluator.cpp

```diff
@@ -35,6 +35,476 @@
 #include "runtime/J9ValueProfiler.hpp"
 #include "env/PersistentCHTable.hpp"
 
+#include "codegen/J9WatchedInstanceFieldSnippet.hpp"
+#include "codegen/J9WatchedStaticFieldSnippet.hpp"
+
+void
+J9::TreeEvaluator::rdWrtbarHelperForFieldWatch(TR::Node *node, TR::CodeGenerator *cg, TR::Register *sideEffectRegister, TR::Register *valueReg)
+   {
+   bool isWrite = node->getOpCode().isWrtBar();
+
+   TR_ASSERT_FATAL(J9ClassHasWatchedFields >= std::numeric_limits<uint16_t>::min() && J9ClassHasWatchedFields <= std::numeric_limits<uint16_t>::max(), "Expecting value of J9ClassHasWatchedFields to be within 16 bits. Currently it's %d(%p).", J9ClassHasWatchedFields, J9ClassHasWatchedFields);
+   // Populate a data snippet with the required information so we can call a VM helper to report the fieldwatch event.
+   J9Method *owningMethod = reinterpret_cast<J9Method *>(node->getOwningMethod());
+   int32_t bcIndex = node->getByteCodeInfo().getByteCodeIndex();
+   TR::SymbolReference *symRef = node->getSymbolReference();
+   bool isUnresolved = symRef->isUnresolved();
+   TR::Snippet *dataSnippet;
```

Comment:
Can we `NULL` initialize such pointer variables? It protects any path following the code from initializing the pointer and having undefined behavior later on (not saying such is the case here but it is good practice to default initialize forward declared variables).

Reply:
f0eae7d